--- a/espressif__esp_rcp_update/CMakeLists.txt
+++ b/espressif__esp_rcp_update/CMakeLists.txt
@@ -1,6 +1,6 @@
 idf_component_register(SRC_DIRS src
                        INCLUDE_DIRS include
-                       REQUIRES driver esp-serial-flasher nvs_flash)
+                       REQUIRES driver esp-serial-flasher esp_driver_gpio esp_driver_uart nvs_flash)
 
 idf_build_get_property(python PYTHON)
 if(CONFIG_AUTO_UPDATE_RCP)
